src/Sarif.Multitool/MergeCommand.cs

```diff
@@ -32,14 +32,20 @@ public static int Run(MergeOptions mergeOptions)
                     combinedLog = new InsertOptionalDataVisitor(dataToInsert).VisitSarifLog(combinedLog);
                 }
 
+                string outputDirectory = mergeOptions.OutputFolderPath ?? Environment.CurrentDirectory;
+
                 // Write output to file.
-                string outputName = Path.Combine(mergeOptions.OutputFolderPath, GetOutputFileName(mergeOptions));
+                string outputName = Path.Combine(outputDirectory, GetOutputFileName(mergeOptions));
                 
                 var formatting = mergeOptions.PrettyPrint
                     ? Formatting.Indented
                     : Formatting.None;
 
-                Directory.CreateDirectory(mergeOptions.OutputFolderPath);
+                if (!Directory.Exists(outputDirectory))
+                {
+                    Directory.CreateDirectory(mergeOptions.OutputFolderPath);
+                }
```

Comment:
>} [](start = 16, length = 1)

Oh wow. I think I understand that you're doing this as a micro-optimization, but the code is a bit unclear.

It works because if the calculated `outputDirectory` doesn't exist, it must have been specified in the `mergeOptions` -- because otherwise, `outputDirectory` would be the current working directory, and that _must_ exist. Therefore it is safe to say, as you do here: "if the output directory doesn't exist, get it from the options." If you'd just written "if the output directory doesn't exist, create it", I wouldn't have given it a second thought.

Finally, since `CreateDirectory` is idempotent, you might just as well call it. I don't see the savings as worth the extra line of code. I've just spent more time thinking about this then we'll save in CPU time until the heat death of the universe. Was it worth it? ;-)

Reply:
@michaelcfanning Won't we crash now if `OutputFolderPath` is not specified? Shouldn't this be `outputDirectory`?

---
In reply to: [214520177](https://github.com/Microsoft/sarif-sdk/pull/1062#discussion_r214520177) [](ancestors = 214520177)